api/_db.data-utils.js

```diff
@@ -53,6 +53,40 @@ const parseDate = (rawValue) => {
 	return parsed;
 };
 
+const findBestCertificate = (certificates) => {
+	//The best certificate is found as follows:
+	// 1. Last issued RED card (that is still valid)
+	// 2. Last issued GREEN card (that is still valid)
+	// 3. Any expired card
+	// 4. Any invalid card
+
+	let bestRedCard = null;
+	let bestGreenCard = null;
+	let anyExpiredCard = null;
+	let anyInvalidCard = null;
+
+	for (const card of certificates) {
+		const expiryDate = parseDate(card.expiryTime);
+		const examDate = parseDate(card.examTime);
+
+		if ((card.certificate !== CODE.RED && card.certificate !== CODE.GREEN) || !expiryDate) {
```

Comment:
`!expiryDate` check semantically belongs to the next `if` and since you're then later checking `card.certificate === CODE.{GREEN,RED}` anyway, I'd replace this whole block with `else` in the end.

Reply:
We need to check  for cards with `expiryDate===null` right at the beginning so that cards without it would be filtered out right away: otherwise these would fall into `expiryDate < Date.now()` and we need to differentiate invalid and expired cards. I refactored it a bit so that the logic would be clearer.